envconfig_test.go

```diff
@@ -70,6 +75,7 @@ type EmbeddedButIgnored struct {
 
 func TestProcess(t *testing.T) {
 	var s Specification
+	s.NestedInterface = &Anonymous{}
```

Comment:
Yeah, I tried to explains the low-level version of my use case. but in real-life my use case is more complicated.

I could only achieve this with an `interface{}`